Server/Item.cs

```diff
@@ -2415,7 +2415,16 @@ public Map Map
 
 					OnMapChange();
 
-					if (old == null || old == Map.Internal)
+                    if (Sockets != null)

```

Comment:
Are these core edits still needed?

Reply:
Yes to call his new method OnMapChange. You could also handle this in the item itself (override it) instead of a core change.

Reply:
Ahh this was already done before I commented, very nice catch. I go by a rule, where if 2 or more derived classes could benefit from making a virtual function, property, etc, then i make the core edit, or if there would be practical use for those who like to make custom scripts.